project-base/storefront/components/Blocks/Popup/ManageCustomerUserPopup.tsx

```diff
@@ -0,0 +1,248 @@
+import { useCurrentCustomerData } from 'connectors/customer/CurrentCustomer';
+import { useAddNewCustomerUserMutation } from 'graphql/requests/customer/mutations/AddNewCustomerUserMutation.generated';
+import { useEditCustomerUserPersonalDataMutation } from 'graphql/requests/customer/mutations/EditCustomerUserPersonalDataMutation.generated';
+import { TypeCompanyCustomerUser } from 'graphql/types';
+import { GtmMessageOriginType } from 'gtm/enums/GtmMessageOriginType';
+import useTranslation from 'next-translate/useTranslation';
+import { Controller, FormProvider, Path, SubmitHandler, UseFormReturn } from 'react-hook-form';
+import { useSessionStore } from 'store/useSessionStore';
+import { CustomerUserManageProfileFormType } from 'types/form';
+import { CombinedError } from 'urql';
+import { useCurrentCustomerUserPermissions } from 'utils/auth/useCurrentCustomerUserPermissions';
+import { getUserFriendlyErrors } from 'utils/errors/friendlyErrorMessageParser';
+import { useErrorPopup } from 'utils/forms/useErrorPopup';
+import { showErrorMessage } from 'utils/toasts/showErrorMessage';
+import { showSuccessMessage } from 'utils/toasts/showSuccessMessage';
+import { useCustomerUserGroupsAsSelectOptions } from 'utils/user/useCustomerUserGroupsAsSelectOptions';
+
+type ManageCustomerUserPopupProps = {
+    customerUser?: TypeCompanyCustomerUser;
+    mode?: 'edit' | 'add';
+};
+
+export const ManageCustomerUserPopup: FC<ManageCustomerUserPopupProps> = ({ customerUser, mode = 'edit' }) => {
+    const { t } = useTranslation();
+    const [, customerEditUser] = useEditCustomerUserPersonalDataMutation();
+    const [, customerAddUser] = useAddNewCustomerUserMutation();
+    const updatePortalContent = useSessionStore((s) => s.updatePortalContent);
+    const { canManageProfile } = useCurrentCustomerUserPermissions();
+    const customerUserRoleGroupsAsSelectOptions = useCustomerUserGroupsAsSelectOptions();
+    const currentCustomerUserData = useCurrentCustomerData(customerUser);
+
+    const customerUserDefaultFormData =
+        customerUser && currentCustomerUserData
+            ? {
+                  ...currentCustomerUserData,
+                  roleGroup: {
+                      label: currentCustomerUserData.roleGroup.name,
+                      value: currentCustomerUserData.roleGroup.uuid,
+                  },
+              }
+            : {
+                  email: '',
+                  firstName: '',
+                  lastName: '',
+                  telephone: '',
+                  roleGroup: {
+                      label: '',
+                      value: '',
+                  },
+              };
+
+    const [formProviderMethods] = useCustomerUserManageProfileForm(customerUserDefaultFormData);
+    const formMeta = useCustomerUserManageProfileFormMeta(formProviderMethods, mode);
+
+    useErrorPopup(formProviderMethods, formMeta.fields, undefined, GtmMessageOriginType.other);
```

Comment:
Here's a more polished response:

---

I recommend removing the error popup, as it negatively impacts the user experience when submitting a form with missing required fields. Currently, the process closes the existing popup and opens a new one displaying the errors, causing the user to lose all their inputs. Instead, we should consider displaying inline validation messages.

Reply:
Yes I haven't notice this but you are right, this is bad for the UX. Removed the error popup.